modules/text2vec-openai/vectorizer/objects.go

```diff
@@ -99,3 +118,113 @@ func (v *Vectorizer) getVectorizationConfig(cfg moduletools.ClassConfig) ent.Vec
 		Dimensions:   settings.Dimensions(),
 	}
 }
+
+// batchWorker is a go routine that handles the communication with the vectorizer
+//
+// On the high level it has the following steps:
+//  1. It receives a batch job
+//  2. It splits the job into smaller vectorizer-batches if the token limit is reached. Note that objects from different
+//     batches are not mixed with each other.
+//  3. It sends the smaller batches to the vectorizer
+func (v *Vectorizer) batchWorker() {
+	tokenLimit := 1000 // don't know limits before the first request
+	remainingRequests := 10
+	requestLimitReset := 0
+	texts := make([]string, 0, 100)
+	for job := range v.jobQueueCh {
+		conf := v.getVectorizationConfig(job.cfg)
+		objCounter := 0
+		vecBatchOffset := 0 // index of first object in current vectorizer-batch
+
+		tokensInCurrentBatch := 0
+		texts = texts[:0]
+		for {
+			// add input to vectorizer-batches until current token limit is reached
+			text := job.texts[objCounter]
+			tokensInCurrentBatch += job.tokens[objCounter]
+			if float32(tokensInCurrentBatch) < 0.95*float32(tokenLimit) {
+				texts = append(texts, text)
+				if objCounter < len(job.objects)-1 {
+					objCounter++
+					continue
+				}
+			}
+
+			if _, ok := job.ctx.Deadline(); !ok {
```

Comment:
From docs `Deadline returns ok==false when no deadline is set`. Is that intended usage?
Shouldn't be `ctx.Err()`? Existent error indicates context being cancelled or expired.

Reply:
yep, thanks!